sdk/resourcemanager/Azure.ResourceManager/src/ArmResource.cs

```diff
@@ -31,74 +33,48 @@ protected ArmResource()
-        /// <param name="credential"></param>
-        /// <param name="uri"></param>
-        /// <param name="pipeline"></param>
-        /// <param name="id"> The identifier of the resource that is the target of operations. </param>
-        protected ArmResource(ArmClientOptions clientOptions, TokenCredential credential, Uri uri, HttpPipeline pipeline, ResourceIdentifier id)
-            : this(new ClientContext(clientOptions, credential, uri, pipeline), id)
-        {
-        }
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="ArmResource"/> class.
-        /// </summary>
-        /// <param name="clientContext"></param>
-        /// <param name="id"> The identifier of the resource that is the target of operations. </param>
-        internal ArmResource(ClientContext clientContext, ResourceIdentifier id)
+        protected internal ArmResource(ArmClient armClient, ResourceIdentifier id)
         {
             Argument.AssertNotNull(id, nameof(id));
 
-            ClientOptions = clientContext.ClientOptions;
+            ArmClient = armClient;
             Id = id;
-            Credential = clientContext.Credential;
-            BaseUri = clientContext.BaseUri;
-            Pipeline = clientContext.Pipeline;
         }
 
-        private Tenant Tenant => _tenant ??= new Tenant(ClientOptions, Credential, BaseUri, Pipeline);
+        private Tenant Tenant => _tenant ??= new Tenant(ArmClient);
 
         /// <summary>
-        /// Gets the resource identifier.
+        /// Gets the <see cref="ArmClient"/> this resource client was created from.
         /// </summary>
-        public virtual ResourceIdentifier Id { get; }
+        protected internal virtual ArmClient ArmClient { get; }
 
         /// <summary>
-        /// Gets the Azure Resource Manager client options.
+        /// Gets the diagnostic options for this resource client.
         /// </summary>
-        protected internal virtual ArmClientOptions ClientOptions { get; private set; }
+        protected internal DiagnosticsOptions DiagnosticOptions => ArmClient.DiagnosticOptions;
 
         /// <summary>
-        /// Gets the Azure credential.
+        /// Gets the pipeline for this resource client.
         /// </summary>
-        protected internal virtual TokenCredential Credential { get; private set; }
+        protected internal HttpPipeline Pipeline => ArmClient.Pipeline;
 
         /// <summary>
-        /// Gets the base URI of the service.
+        /// Gets the base uri for this resource client.
         /// </summary>
-        protected internal virtual Uri BaseUri { get; private set; }
+        protected internal Uri BaseUri => ArmClient.BaseUri;
 
         /// <summary>
-        /// Gets the HTTP pipeline.
+        /// Gets the resource identifier.
         /// </summary>
-        protected internal virtual HttpPipeline Pipeline { get; }
+        public virtual ResourceIdentifier Id { get; }
```

Comment:
does this have to be virtual? We pass the value into the ctor

Reply:
needed for test instrumentation